source/elements/oneDAL/source/common_interface/error_handling.rst

```diff
@@ -1,3 +1,131 @@
+.. _error_handling:
+
 ==============
 Error handling
 ==============
+
+oneDAL error handling relies on the  mechanism of C++ exceptions. If an error
+occurs, it shall be propagated at the point of a function call where it can be
+caught using standard C++ error handling mechanism.
+
+Exception classification
+========================
+
+Exception classification in onDAL is aligned with C++ STL classification. oneDAL
+shall introduce abstract classes defining the base for for hierarchy of
+exception classes. Сoncrete exception classes are derived from respective STL
+exception classes. oneDAL library shall throw exceptions represented with
+concrete classes.
+
+The root of exception hierarchy is ``onedal::exception``. The rest exception
+classes are derived from this class.
+
+::
+
+    class onedal::exception;
+
+.. list-table::
+   :widths: 30 65 5
+   :header-rows: 1
+
+   * - Exception
+     - Description
+     - Abstract
+   * - ``onedal::exception``
+     - Base class of oneDAL exception hierarchy.
+     - Yes
+
+All errors divide into 3 groups: logic errors, runtime errors and bad allocation.
```

Comment:
divided, the use of bullets is better
All oneDAL exceptions shall be divided into three groups:
- logic
- runtime
- bad allocation - but it is confusing as a group